code/modules/mob/living/simple_animal/friendly/corgi.dm

```diff
@@ -498,6 +504,10 @@
 /mob/living/simple_animal/corgi/proc/remove_inventory(var/remove_from = "head", mob/user)
 	switch(remove_from)
 		if("head")
+			if(!can_remove_head)
```

Comment:
Maybe a more user-friendly response would be to remove the entire hardsuit when you try to remove the helmet from the corgi, with appropriate feedback message. At least that's my opinion.